### PR TITLE
Add CLI option to override shell used

### LINKED ISSUE #117: Add ability to override shell used
I.e. if I were to try and run `demo` with an Alpine-based distro, I receive the error:

```
2023/09/23 13:51:56 run failed: step command failed: exec: "bash": executable file not found in $PATH
exit status 1
```

### PR DESCRIPTION
**Pull request by jamietanna**
- Default to `sh` for `Ensure`
- Add CLI option to override shell used

Closes #117.

### CODE REVIEW

File: run.go [hunk 1]
```diff
@@ -207,7 +209,7 @@ func (s *step) echo(current, max int) {
 
 func (s *step) execute() error {
 	joinedCommand := strings.Join(s.command, " ")
-	cmd := exec.Command(bash, "-c", joinedCommand)
+	cmd := exec.Command(s.r.options.Shell, "-c", joinedCommand)
```
Comment (saschagrunert) [inline run.go:212]: Hm, we assume that every shell has a `-c` command, but I think that's fine for now.

**Review by saschagrunert — APPROVED** [review]

**Comment by saschagrunert** [comment]
The lint CI is not happy unfortunately.

File: run.go (hunk 1 above)
Comment (jamietanna) [reply run.go:212]: I believe this is a common thing - I've checked `sh`, `bash`, `ksh`, `zsh` and `fish` and they all use it

**Review by saschagrunert — APPROVED** [review]

**Comment by saschagrunert** [comment]
The linter now complains about the line length :upside_down_face:

**Comment by jamietanna** [comment]
Sorry yeah just trying to sort - unfortunately my local Go version isn't compatible with `golangci-lint` hence needing the pushes - sorry about that!

**Comment by jamietanna** [comment]
Thank you :rocket:

**Pull request merged**